angular/src/lib/drawer/drawer.component.ts

```diff
@@ -0,0 +1,60 @@
+ /** @component drawer */
+import { Component, Input, Output, EventEmitter, HostBinding } from '@angular/core';
+
+export type DrawerOpenFromDirection = 'right' | 'top' | 'bottom';
+export type DrawerSize = 'medium' | 'large';
+
+@Component({
+  selector: 'md-drawer',
+  template: `
+    <ng-content></ng-content>
+    <md-icon className="md-drawer__close" name="cancel_14" ariaLabel="{{closeAriaLabel}}" (click)="close()" ></md-icon>
+  `,
+  styles: []
+})
+export class DrawerComponent {
+
+  private _visible: Boolean = false;
+  public color: string = 'blue';
+
+
+  /** @option css class names | '' | '' */
+  @Input() public class: string = '';
+
+  @HostBinding('class') get className(): string {
+    // tslint:disable-next-line:max-line-length
+    return `${this.class ? this.class : '' } md-drawer ${!this.isOpen ? 'md-drawer--closed' : ''} ${'md-drawer--' + this.openFrom} ${'md-drawer--' + this.size }`;
+  }
+
+   /** @prop Set whether the drawer is open or not defaults to false (closed) */
+  @Input()
+  set isOpen (value: Boolean) {
+    this._visible = value;
+
+  }
+  get isOpen() {
+    return this._visible;
+  }
+
+   /** @prop Sets the openFrom opens  defaults to right */
+  @Input() openFrom: DrawerOpenFromDirection = 'right';
+
+   /** @prop Sets the openFrom opens defaults to medium  */
+   @Input() size: DrawerSize = 'medium';
+
+   /** @prop set the text for accessibilty on the close button */
+   @Input() closeAriaLabel: String = 'close';
+
+  /** use this event to know when the drawer has changed */
+  @Output() drawerChanged: EventEmitter<Boolean> = new EventEmitter();
+
+  close() {
+    this.isOpen = false;
+    this.drawerChanged.emit(this.isOpen);
+  }
+  constructor() {
```

Comment:
@daanishnasir is this necessary for an empty constructor? In react we leave it out if it isn't necessary.  If so, can we determine a code style?

Reply:
not necessary to have if unused